BankLibrary/Bank.cs

```diff
@@ -9,18 +10,74 @@ public class Bank<T> where T : Account
 
         public void OpenAccount(OpenAccountParameters parameters)
         {
-            // TODO: check types compatibility
-            CreateAccount(parameters.AccountCreated, () => parameters.Type == AccountType.Deposit 
+            CheckValidTypeAccount(parameters.Type);
+
+            CreateAccount(parameters, () => parameters.Type == AccountType.Deposit
                 ? new DepositAccount(parameters.Amount) as T
                 : new OnDemandAccount(parameters.Amount) as T);
         }
 
-        private void CreateAccount(AccountCreated accountCreated, Func<T> creator)
+        public void Withdraw(decimal amount, int id)
+        {
+            var account = GetAccountById(id);
+
+            if (account is DepositAccount depositAccount)
+            {
+                depositAccount.Withdraw(amount);
+            }
+            else
+            {
+                account.Withdraw(amount);
+            }
+        }
+
+        public void Put(decimal amount, int id)
+        {
+            var account = GetAccountById(id);
+            account.Put(amount);
+        }
+
+        public void Close(int id)
+        {
+            var account = GetAccountById(id);
+            account.Close();
+        }
+
+        public void IncrementDays()
+        {
+            _accounts.ForEach(x => x.IncrementDays());
+        }
+
+        private void CreateAccount(OpenAccountParameters parameters, Func<T> creator)
         {
             var account = creator();
+            account.Created += parameters.AccountCreated;
+            account.WithdrawMoney += parameters.WithdrawMoney;
+            account.PutMoney += parameters.PutMoney;
+            account.AccountClosed += parameters.AccountClosed;
             account.Open();
-            account.Created += accountCreated;
             _accounts.Add(account);
         }
+
+        private Account GetAccountById(int id)
+        {
+            var account = _accounts.Where(x => x.Id == id).FirstOrDefault();
```

Comment:
if we expect only one account with a given ID to exist - use `SingleOrDefault` instead
```suggestion
            var account = _accounts.Where(x => x.Id == id).SingleOrDefault();
```